mlflow/recipes/steps/ingest/datasets.py

```diff
@@ -519,6 +519,15 @@ class _SparkDatasetMixin:
     and conversion to parquet format.
     """
 
+    def _convert_spark_df_to_pandas(self, spark_df):
+        import pandas as pd
+
+        datetime_cols = [schema[0] for schema in spark_df.dtypes if schema[1].startswith("date")]
+        pandas_df = spark_df.toPandas()
+        pandas_df[datetime_cols] = pandas_df[datetime_cols].apply(pd.to_datetime, errors="coerce")
```

Comment:
@HyukjinKwon Curious why pyspark does not do the conversion in `toPandas`

Reply:
If this is to convert regular `datetime.date`s to `datetime64[ns]`, it was intentional because `datetime64[ns]` can contain more types including time that is matched with `TimestampType` in Spark.

There is no exact matching for date itself so we use `datetime.date` with `object` type.